dali/kernels/common/copy.h

```diff
@@ -55,6 +55,86 @@ void copy(const TensorView<StorageOut, TOut, NDimIn>& out,
   copy<StorageOut, StorageIn>(out.data, in.data, in.num_elements() * sizeof(TOut), stream);
 }
 
+/**
+ * @brief Copy in to out, merging contiguous samples.
+ *
+ * Contiguous samples are merged to reduce number of copies issuead and, at the same time,
+ * allow for non-contiguous data.
+ * The function does not do full coalescing - samples are not sorted to discover order
+ *
+ * This function has very relaxed shape verification - samples that are stored in contiguous
+ * memory are merged and those groups need to have the same total length. Empty samples are skipped
+ * and are not treated as discontinuities.
+ */
+template <typename StorageOut, typename TOut, int NDimOut,
+          typename StorageIn, typename TIn, int NDimIn>
+void copy(const TensorListView<StorageOut, TOut, NDimOut> &out,
+          const TensorListView<StorageIn, TIn, NDimIn> &in,
+          cudaStream_t stream = 0) {
+  static_assert(sizeof(TOut) == sizeof(TIn), "Tensor elements must be of equal size!");
```

Comment:
So you are prohibiting reinterpreting data during copy, right?

Reply:
Currently yes, but it's a good idea to allow it, actually.

Reply:
Suggestion. Up to you.